File: practice-01/exercise-03/go/main.go
package main

import "fmt"

const N = 3

type Student struct {
	name     string
	lastname string
	dni      int
}

func load(x *[N]Student) {
	for i := 0; i < N; i++ {
		fmt.Print("Nombre: ")
		fmt.Scan(&x[i].name)
		fmt.Print("Apellido: ")
		fmt.Scan(&x[i].lastname)
		fmt.Print("DNI: ")
		fmt.Scan(&x[i].dni)
	}
}

func show(x *[N]Student) {
	for _, s := range x {
		fmt.Printf("nombre: %s, apellido: %s, dni: %d\n", s.name, s.lastname, s.dni)
	}
}

func count(x *[N]Student) {
	var count = 0
	for _, s := range x {
		if s.dni > 40_000_000 {
			count++
		}
	}
	fmt.Printf("Cantidad de DNIs mayores a 40M: %d\n", count)
}

func main() {
	var students [N]Student
	load(&students)
	fmt.Println()
	show(&students)
	fmt.Println()
	count(&students)
}

File: practice-01/exercise-03/c/main.c
#include <stdio.h>
#include <strings.h>
#define N 3
#define M 20

struct Student
{
  char name[M];
  char lastname[M];
  int dni;
};

void load(struct Student x[N])
{
  for (int i = 0; i < N; i++)
  {
    printf("Nombre: ");
    fgets(x[i].name, sizeof(char) * M, stdin);
    printf("Apellido: ");
    fgets(x[i].lastname, sizeof(char) * M, stdin);
    printf("Dni: ");
    scanf("%d", &x[i].dni);
    while (getchar() != '\n');
  }
}

void show(struct Student x[N])
{
  for (int i = 0; i < N; i++)
    printf("nombre: %s apellido: %s dni: %d\n", x[i].name, x[i].lastname, x[i].dni);
}

void count(struct Student x[N])
{
  int count = 0;
  for (int i = 0; i < N; i++)
  {
    if (x[i].dni > 40000000)
      count++;
  }
  printf("Cantidad de DNIs mayores a 40M: %d", count);
}

void main()
{
  struct Student students[N];
  load(students);
  printf("\n");
  show(students);
  printf("\n");
  count(students);
}